hypha/apply/activity/messaging.py

```diff
@@ -716,23 +716,28 @@ def slack_channels(self, source, **kwargs):
     def send_message(self, message, recipient, source, **kwargs):
         target_rooms = self.slack_channels(source, **kwargs)
 
-        if not self.destination or not any(target_rooms):
+        if not self.destination or not any(target_rooms) or not settings.SLACK_TOKEN:
             errors = list()
             if not self.destination:
                 errors.append('Destination URL')
             if not target_rooms:
                 errors.append('Room ID')
+            if not settings.SLACK_TOKEN:
+                errors.append('Slack Token')
             return 'Missing configuration: {}'.format(', '.join(errors))
 
         message = ' '.join([recipient, message]).strip()
 
         data = {
-            "room": target_rooms,
             "message": message,
         }
-        response = requests.post(self.destination, json=data)
-
-        return str(response.status_code) + ': ' + response.content.decode()
+        for room in target_rooms:
+            try:
+                slack_message('messages/slack_message.html', data, channel=room)
```

Comment:
@sandeepsajan0 so we don't have to initialise django slack with the env variables and it automatically uses them from the settings?

Reply:
No, we don't need to initialise it. It automatically uses SLACK_TOKEN, and other env variables to send the messages.